numpy/lib/_function_base_impl.py

```diff
@@ -3832,6 +3838,8 @@ def median(a, axis=None, out=None, overwrite_input=False, keepdims=False):
     array([6.5, 4.5, 2.5])
     >>> np.median(a, axis=1)
     array([7.,  2.])
+    >>> np.median(b, axis=(0,1), overwrite_input=True)
+    3.5
```

Comment:
These examples in the docstrings should be runnable. Using `b` here when it hasn't been defined above doesn't make sense. I would also not use `overwrite_input`, since that confuses things. When I use `np.median` with `a` defined above on the latest version of numpy dev, I get the result above.

In principle doctests should catch these, but I'm not sure we run doctests on the full numpy codebase...

Instead, I think it makes more sense to just rewrite all of these examples to use a small 3D array, so this example where you take the median successively over two dimensions makes sense. Something like:

```python
>>> import numpy as np
>>> a = np.arange(27)
>>> a.shape = (3, 3, 3)
>>> a
array([[[ 0,  1,  2],
        [ 3,  4,  5],
        [ 6,  7,  8]],

       [[ 9, 10, 11],
        [12, 13, 14],
        [15, 16, 17]],

       [[18, 19, 20],
        [21, 22, 23],
        [24, 25, 26]]])
>>> np.median(a)
np.float64(13.0)
>>> np.median(a, axis=0)
array([[ 9., 10., 11.],
       [12., 13., 14.],
       [15., 16., 17.]])
>>> np.median(a, axis=(0, 1))
array([12., 13., 14.])
```

And then if you didn't feel like editing the rest of this, you could redefine a to its original value below this example.

Just a thought, though! There are probably other ways to edit this that would make sense.

Also it would be nice if you could make sure this docstring is actually fully runnable and updated, since it looks like it hasn't been updated for the fallout from NEP 50 (ping @seberg - there's probably a lot of other docstrings that need to be updated right?).

Reply:
Err, not NEP 50, I guess just the new reprs for scalars?

Reply:
Yeah, that is on my not-done list.  I had always hoped to use https://github.com/scientific-python/pytest-doctestplus/pull/227 although that is a bit tedious until we better support doctestplus.

OTOH, a few failures (and misses) shouldn't matter too much.

At this time, I think just don't worry about whether or not you write `np.float64(1.0)` or `1.0`, keeping things consistent may be better...